File: 121.best-time-to-buy-and-sell-stock.kt
// @leet start
class Solution {
    fun maxProfit(prices: IntArray): Int {
      var min = prices[0]
      var max = 0
      for (i in prices.indices) {
        if (prices[i] < min) {
          min = prices[i]
        }
        if (prices[i] - min > max) {
          max = prices[i] - min
        }
      }
      return max
    }
}
// @leet end

File: 121.best-time-to-buy-and-sell-stock.cs
// @leet start
public class Solution {
    public int MaxProfit(int[] prices) {
      var min = prices[0];
      var max = 0;
      for (int i = 1; i < prices.Length; i++) {
        if (prices[i] < min) {
          min = prices[i];
        }
        if (prices[i] - min > max) {
          max = prices[i] - min;
        }
      }
      return max;
    }
}
// @leet end
